### DISCUSSION TITLE
Battery Backup for clean shutdown

### DISCUSSION THREAD

**Discussion by mkegruber** [akuker/RASCSI#1458] [Ideas]
Just a thought,

What about a mod that features an onboard lithium battery + recharging circuit so that when the host device is on the lithium battery is being charged, and when the host device is powered power fails over to the onboard battery backup while a shutdown signal / command is set to the raspberry pi for a clean shutdown to mitigate file corruption due to shutdown power off(?)

**Comment by rdmark** [comment]
It's a great idea. :)

I think you already got the response over on Discord, but for the record here it is: https://github.com/akuker/SuperUPS

SuperUPS does pretty much what you're asking here, except it uses super capacitors rather than lithium cells.
